liblwgeom/ptarray.c

```diff
@@ -1498,11 +1499,14 @@ ptarray_remove_repeated_points_in_place(POINTARRAY *pa, double tolerance, uint32
 			if (last_point && n_points_out > 1 && tolerance > 0.0 && dsq <= tolsq)
 			{
 				n_points_out--;
+				p_to -= pt_size;
 			}
 		}
 
 		/* Compact all remaining values to front of array */
-		ptarray_copy_point(pa, i, n_points_out++);
```

Comment:
can this be done by, say, moving the ptarray_copy_point into lwinline.c? should render to similar assembly if inlined, and worth investigating if not.

Reply:
> can this be done by, say, moving the ptarray_copy_point into lwinline.c? should render to similar assembly if inlined, and worth investigating if not.

I've tested it with https://github.com/Algunenano/postgis/commit/9508c2d8e870ee895996bf05de91f1187e8996a0 and it's slower than the approach in this PR:

```
number of transactions actually processed: 73
latency average = 275.065 ms
```
```
number of transactions actually processed: 307
latency average = 65.338 ms
```

But it might be useful for other cases.